### DISCUSSION TITLE
Other controllers than XBOX ones

### DISCUSSION THREAD

**Discussion by 2phenix** [pybricks/support#1458] [Q&A] [answered]
Hello, I saw that it was possible to use an XBOX controller but is it possible to connect another controller like a PlayStation one?

**Comment by laurensvalk** [comment, marked as answer]
The Xbox controller works because it uses Bluetooth Low Energy, which is what the LEGO hubs use. 

Microsoft has released updates to enable BLE on all their Bluetooth controllers.

As far as I know, the Nintendo Switch Controller and Playstation Controllers don't have this or haven't enabled it.

**Reply by 2phenix** [reply]
Ok thanks

**Reply by TheTechGuy13** [reply]
What if you guys somehow programmed or built some sort of adapter for this? That way, others that can't get an Xbox controller can use the others they already have.

**Reply by laurensvalk** [reply]
I think you can do exactly that with this board: https://www.antonsmindstorms.com/product/wifi-python-esp32-board-for-mindstorms/

**Reply by TheTechGuy13** [reply]
sweet thanks. Is there a way I can get my hands on this at a store of some sort?

**Reply by ste7anste7an** [reply]
At antonsmindstorms.com you can order the lms-esp32 board online. We provide a firmware that allows you to hook up any controller (Nintendo, PS4 and xbox) to the LMS-ESP32. The esp32 communicates with the hub that runs either pybricks or the original spike firmware. Here is an overview: https://www.antonsmindstorms.com/2025/04/05/lms-esp32-firmware-options-for-game-pads-servos-and-neopixels-with-bluepad/

**Reply by TheTechGuy13** [reply]
ooh, yay thanks. I may consider buying this sometime in the future.
